AppBundles/DrawingsListPlugin/DrawingsListAutomation.cs

```diff
@@ -50,21 +50,47 @@ public void RunWithArguments(Document doc, NameValueMap map)
             using (new HeartBeat())
             {
                 var rootDir = System.IO.Directory.GetCurrentDirectory();
+
+                if (doc == null)
+                {
+                    doc = inventorApplication.Documents.Open(map.Item["_1"]);
+                }
+
+                var fullFileName = doc.FullFileName;
+                var fileName = System.IO.Path.GetFileNameWithoutExtension(fullFileName);
+
                 var drawingExtensions = new List<string> { ".idw", ".dwg" };
                 var oldVersion = @"oldversions\";
                 var drawings = System.IO.Directory.GetFiles(rootDir, "*.*", System.IO.SearchOption.AllDirectories)
                                     .Where(file => drawingExtensions.IndexOf(System.IO.Path.GetExtension(file.ToLower())) >= 0 &&
                                     !file.ToLower().Contains(oldVersion));
 
                 var index = System.IO.Path.Combine(rootDir, "unzippedIam").Length + 1;
-                drawings = drawings.Select(path => path.Substring(index));
+                // order by filename
+                drawings = drawings.Select(path => path.Substring(index)).OrderBy(d => System.IO.Path.GetFileName(d));
+
+                var defaultDrawings = drawings.Where(d => System.IO.Path.GetFileNameWithoutExtension(d) == fileName).ToArray();
+                var defaultDrawing = defaultDrawings.Length > 0 ? defaultDrawings[0] : null;
+                LogTrace("DEFAULT drawing is: " + defaultDrawing);
+
+                var outputDrawingsList = new List<string>();
+
+                // the first one is default when found
```

Comment:
Seems the whole snippet can be rewritten with changing comparer in OrderBy (since what we want is to put default doc at the top).  Something like:
```csharp
class DefaultDocComparer : IComparer<string>
{
   DefaultDocComparer(string defaultDoc) { _defaultDoc = defaultDoc; }
   public int Compare(string x, string y)
   {
      // special handling for default filename
      if (x == _defaultDoc) return int.MaxValue;
      if (y == _defaultDoc) return int.MinValue;
      return x - y;
   }
}
...
drawings = drawings
                 .Select(path => path.Substring(index))
                 .OrderBy(d => System.IO.Path.GetFileName(d), new DefaultDocComparer(fileName));
```

// it's not tested pseudo-code.... just to show the idea.
// need to adjust it to take care of case-insensitivity

Reply:
I will check it when I did merge from master to branch